rechunker/executors/dask.py

```diff
@@ -0,0 +1,114 @@
+import uuid
+from typing import Iterable
+
+import dask
+import dask.array
+from dask.delayed import Delayed
+from dask.optimization import fuse
+
+from rechunker.types import CopySpec, StagedCopySpec, Executor
+
+
+class DaskExecutor(Executor[Delayed]):
+    """An execution engine based on dask.
+
+    Supports zarr and dask arrays as inputs. Outputs must be zarr arrays.
+
+    Execution plans for DaskExecutors are dask.delayed objects.
+    """
+
+    def prepare_plan(self, specs: Iterable[StagedCopySpec]) -> Delayed:
+        return _staged_copy(specs)
+
+    def execute_plan(self, plan: Delayed, **kwargs):
+        return plan.compute(**kwargs)
```

Comment:
Before, when we subclassed Delayed, we could call `persist` on the Recunked object. That was convenient if you wanted to avoid blocking. Now our only option is `compute`. I'm fine with this change here, but maybe we should think about how to bring back that capability in the future.

Reply:
It's exposed as the `.plan` attribute on the Rechunked object. So you could write `rechunk(...).plan.persist()`.

That said, as I think more about how this API would or would not make sense for other schedulers, I wonder if it really makes sense to have an `execute()` method on Rechunked objects at all. The natural way to execute on particular schedulers may not fit neatly into that API, so perhaps it's better to encourage users to work with the scheduler specific objects representing delayed execution graphs (e.g., `dask.delayed` or `apache_beam.PTransform`).